### PR TITLE
Added Support for EBS Snapshots and Volumes

### PR DESCRIPTION
**Pull request by kedarghule**
This PR adds support for EBS Snapshots and Volumes and populates their nodes in graph DB.

Relationships added:

`(AWSAccount)-[RESOURCE]->(EBSVolume)`
  
`(EBSSnapshot)-[CREATED_FROM]->(EBSVolume)`
  
`(EBSVolume)-[ATTACHED_TO]->(EC2Instance)`

`(AWSAccount)-[RESOURCE]->(EBSSnapshot)`

### CODE REVIEW

**Comment by kedarghule** [comment]
Please review https://github.com/lyft/cartography/pull/640 before reviewing this PR

File: cartography/intel/aws/ec2/snapshots.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    return snapshot_volumes
+
+
+@timeit
+def load_snapshot_volume_relations(
+        neo4j_session: neo4j.Session, data: List[Dict], current_aws_account_id: str, update_tag: int,
+) -> None:
+    ingest_volumes = """
+    UNWIND {snapshot_volumes_list) as volume
+        MERGE (v:EBSVolume{id: volume.VolumeId)
```
Comment (mpurusottamc) [inline cartography/intel/aws/ec2/snapshots.py:74]: @kedarghule There's a syntax error here. a curly bracket `}` is missing. It should be `MERGE (v:EBSVolume{id: volume.VolumeId})`

File: cartography/intel/aws/ec2/snapshots.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    return snapshot_volumes
+
+
+@timeit
+def load_snapshot_volume_relations(
+        neo4j_session: neo4j.Session, data: List[Dict], current_aws_account_id: str, update_tag: int,
+) -> None:
+    ingest_volumes = """
+    UNWIND {snapshot_volumes_list) as volume
```
Comment (mpurusottamc) [inline cartography/intel/aws/ec2/snapshots.py:73]: @kedarghule closing bracket `)` is incorrect here. it should be `}`

File: cartography/intel/aws/ec2/snapshots.py (hunk 1 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/snapshots.py]: Done

File: cartography/intel/aws/ec2/snapshots.py (hunk 2 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/snapshots.py]: Done

File: cartography/intel/aws/ec2/images.py [hunk 3]
```diff
@@ -17,9 +16,9 @@
 @timeit
 @aws_handle_regions
 def get_images(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
-    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/images.py:20]: per the other PR, let's leave all the configs in here, and do that as a separate PR

File: cartography/intel/aws/ec2/instances.py [hunk 4]
```diff
@@ -71,8 +70,7 @@ def load_ec2_instance_network_interfaces(neo4j_session: neo4j.Session, instance_
 
 @timeit
 def load_ec2_instances(
-    neo4j_session: neo4j.Session, data: List[Dict], region: str, current_aws_account_id: str,
-    update_tag: int,
+        neo4j_session: neo4j.Session, data: List[Dict], region: str, current_aws_account_id: str, update_tag: int,
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/instances.py:73]: nothing changed on this line, keep it as is

File: cartography/intel/aws/ec2/instances.py [hunk 5]
```diff
@@ -255,8 +302,8 @@ def cleanup_ec2_instances(neo4j_session: neo4j.Session, common_job_parameters: D
 
 @timeit
 def sync_ec2_instances(
-    neo4j_session: neo4j.Session, boto3_session: boto3.session.Session, regions: List[str], current_aws_account_id: str,
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/instances.py:258]: nothing changed here, leave as is plz

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
Please rebase off main branch once the other PR has been merged.

**Review by mpurusottamc — APPROVED** [review]
Looks good

File: cartography/intel/aws/ec2/images.py (hunk 3 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/images.py:20]: Will do.

File: cartography/intel/aws/ec2/instances.py (hunk 4 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/instances.py:73]: Done

File: cartography/intel/aws/ec2/instances.py (hunk 5 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/instances.py:258]: PyCharm keeps indenting it back as it crosses the 120 characters per line limit.

**Comment by kedarghule** [comment]
All changes made as per review. Lemme know of any more changes.

cc: @achantavy , @FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg , @mpurusottamc

File: cartography/data/jobs/cleanup/aws_import_ec2_instances_cleanup.json [hunk 6]
```diff
@@ -1,53 +1,70 @@
...
-    "iterative": true,
-    "iterationsize": 100
-  },
-  {
-    "query": "MATCH (:EC2Instance)-[r:STS_ASSUMEROLE_ALLOW]->(:AWSRole)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
-    "iterative": true,
-    "iterationsize": 100
-  }],
+  "statements": [
+    {
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/data/jobs/cleanup/aws_import_ec2_instances_cleanup.json:3]: Can you revert the spacing changes here?  You've added some lines for ebs, but it looks like all the other  stuff is the same.

File: cartography/intel/aws/ec2/auto_scaling_groups.py [hunk 7]
```diff
@@ -76,6 +76,8 @@ def load_ec2_auto_scaling_groups(
         SET r.lastupdated = {update_tag}
     """
 
+    # neo4j does not accept datetime objects and values. This loop is used to convert
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/auto_scaling_groups.py:79]: pls remove as file logic is unchanged

File: cartography/intel/aws/ec2/instances.py [hunk 8]
```diff
@@ -71,8 +71,8 @@ def load_ec2_instance_network_interfaces(neo4j_session: neo4j.Session, instance_
 
 @timeit
 def load_ec2_instances(
-    neo4j_session: neo4j.Session, data: List[Dict], region: str, current_aws_account_id: str,
-    update_tag: int,
+        neo4j_session: neo4j.Session, data: List[Dict], region: str, current_aws_account_id: str,
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/instances.py:74]: plz don't change spacing

File: cartography/intel/aws/ec2/instances.py [hunk 9]
```diff
@@ -246,6 +246,55 @@ def load_ec2_instances(
                     ).consume()  # TODO see issue 170
 
             load_ec2_instance_network_interfaces(neo4j_session, instance, update_tag)
+            sync_ec2_instance_ebs_volumes(neo4j_session, instance, current_aws_account_id, update_tag)
+
+
+@timeit
+def sync_ec2_instance_ebs_volumes(
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/instances.py:253]: This is an unnecessary layer of abstraction. Please move the two lines (256-7) into the calling function at line 249.

File: cartography/intel/aws/ec2/instances.py (hunk 5 above)
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply cartography/intel/aws/ec2/instances.py:258]: change it in the config?

Comment (kedarghule) [reply cartography/intel/aws/ec2/instances.py:258]: It's also a linter requirement. I'll fail the linter test if the line exceeds 120 characters per line.

File: cartography/data/jobs/cleanup/aws_import_ec2_instances_cleanup.json (hunk 6 above)
Comment (kedarghule) [reply cartography/data/jobs/cleanup/aws_import_ec2_instances_cleanup.json:3]: Done

File: cartography/intel/aws/ec2/auto_scaling_groups.py (hunk 7 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/auto_scaling_groups.py:79]: Done

File: cartography/intel/aws/ec2/instances.py (hunk 8 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/instances.py:74]: This was again as a result of the linter. For some reason the linter failed due to the previous indentation formatting. Passed for this one.

File: cartography/intel/aws/ec2/instances.py (hunk 9 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/instances.py:253]: Done

**Comment by kedarghule** [comment]
Made necessary changes and added in clarifications where necessary :) Lemme know if there's anything more!

File: cartography/intel/aws/ec2/instances.py (hunk 5 above)
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply cartography/intel/aws/ec2/instances.py:258]: oh ok coool

File: cartography/intel/aws/ec2/images.py [hunk 10]
```diff
@@ -19,7 +19,7 @@
 def get_images(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
     client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
     try:
-        images = client.describe_images(Owners=['self'])['Images']
```
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/aws/ec2/images.py:22]: why remove thtis?

**Review by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg — COMMENTED** [review]
One more q on removing `owners[self]`. @achantavy any comments or are we good after that q is resolved?

File: cartography/intel/aws/ec2/images.py (hunk 10 above)
Comment (kedarghule) [reply cartography/intel/aws/ec2/images.py:22]: Looks like I overlooked this during the PR merge conflicts. Rectified it. My bad. Apologies.

Comment (kedarghule) [reply cartography/intel/aws/ec2/images.py]: Good you pointed this out. I would have undone a fix 🙈

**Review by achantavy — APPROVED** [review]
Looks good, thanks @kedarghule!

**Comment by kedarghule** [comment]
> Looks good, thanks @kedarghule!

You may merge this @achantavy :)
